civicrm.module

```diff
@@ -214,6 +214,17 @@ function civicrm_initialize() {
     // initialize the system by creating a config object
     $config = CRM_Core_Config::singleton();
 
+    // Get the menu items.
+    $args = explode('?', $_GET['q']);
+    $path = $args[0];
+
+    // Get the menu for above URL.
+    $item = CRM_Core_Menu::get($path);
+
+    if (!empty(CRM_Utils_Array::value('is_public', $item))) {
+      $config->userFrameworkFrontend = TRUE;
```

Comment:
The good and the bad:

* The really good thing is that it makes Civi's internal metadata (`$config->userFrameworkFrontend`) more consistent across the various CMS's -- i.e. the flag gets a similar meaning on D7, WP, J, etal. It also allow some cleanup re: https://github.com/civicrm/civicrm-core/pull/12929 so that the theming options are similar in each environment.
* From an `r-tech` perspective, the tricky thing is that there is other functionality (`rgrep userFrameworkFrontend`) which keys off of this variable - and so one needs to give a think to whether D7 folks hitting those linse will perceive that the change as regressive.

Reply:
It would be really nice to get this consistent with other CMSs.  @totten I assume this has been discussed but what does is_public actually mean and how is it set? In other words, is it likely to be reliable?

Reply:
@mattwire is_public is set in the xml/menu routes and i this is generally code for no permissions needed or something

Reply:
@seamuslee001 I would note that we discussed one method we sometimes use to grandfather things in is permitting 'define' - I see this as a setting being something we're expecting to keep as an option whereas a define being something we support to allow the advance guard to test . a change we think we want to make generically but want more real world validation of.